### PR TITLE
Global SystemReports and OutputProcessor

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------
 - Move SystemReports and OutputProcessor to state

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mitchute** [comment]
The failing test runs fine locally for me. I think this may be one of those false-positives that occasionally pop up. Everything else looks fine. Merging.

File: tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc [hunk 1]
```diff
@@ -402,19 +402,16 @@ TEST_F(EnergyPlusFixture, Test_UnitaryHybridAirConditioner_Unittest)
...
+    state->dataSysRpts->ZoneOAVolFlowStdRho.allocate(state->dataGlobal->NumOfZones);
+    state->dataSysRpts->ZoneOAVolFlowCrntRho.allocate(state->dataGlobal->NumOfZones);
+    state->dataSysRpts->ZoneOAVolStdRho.allocate(state->dataGlobal->NumOfZones);
+    state->dataSysRpts->ZoneOAVolCrntRho.allocate(state->dataGlobal->NumOfZones);
+    state->dataSysRpts->ZoneMechACH.allocate(state->dataGlobal->NumOfZones);
     MAT.allocate(state->dataGlobal->NumOfZones);
     ZoneAirHumRatAvg.allocate(state->dataGlobal->NumOfZones);
-    MaxHeatingLoadMetByVent.allocate(state->dataGlobal->NumOfZones);
-    MaxOverheatingByVent.allocate(state->dataGlobal->NumOfZones);
-    MaxCoolingLoadMetByVent.allocate(state->dataGlobal->NumOfZones);
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:416]: @mitchute These lines got lost in the shuffle here. Replacing them should fix the unit test failure.

Comment (mitchute) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:416]: Thanks for the 🦅  👁️  @mjwitte. I'm confused why this didn't fail on my local machine 😕  

I've pushed a fix here: https://github.com/NREL/EnergyPlus/pull/8432/commits/890fd55d26356a38537c1669e1b3aaece84bde6c

Comment (mjwitte) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:416]: @mitchute Is it possible that debug builds on Mac and Linux don't catch array bounds violations? The system report stuff in this unit test where it failed on array bounds just happens to fire, but the results aren't important for the test.

Comment (mitchute) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:416]: I'd have to check when/where the array bounds violations are checked. I just ran it before and after the fix it in release on Mac and it sailed through without complaining both times.

Comment (mjwitte) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:416]: "release" - that explains it. Try a debug build and it should fail.

Comment (mjwitte) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:416]: No bounds checking in release

Comment (mitchute) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:416]: Got it. Confirmed that it does fail in debug locally. Thanks, @mjwitte!

**Pull request merged**